src/IonEventStream.ts

```diff
@@ -0,0 +1,410 @@
+                            break;
+                        case IonTypes.STRING:
+                            writer.writeString(tempEvent.ionValue, tempEvent.annotations);
+                            break;
+                        case IonTypes.SYMBOL :
+                            writer.writeSymbol(tempEvent.ionValue, tempEvent.annotations);
+                            break;
+                        case IonTypes.INT :
+                            writer.writeInt(tempEvent.ionValue, tempEvent.annotations);
+                            break;
+                        case IonTypes.DECIMAL :
+                            writer.writeDecimal(tempEvent.ionValue, tempEvent.annotations);
+                            break;
+                        case IonTypes.FLOAT :
+                            writer.writeFloat32(tempEvent.ionValue, tempEvent.annotations);
+                            break;
+                        case IonTypes.NULL :
+                            writer.writeNull(tempEvent.ionType.bid, tempEvent.annotations);
+                            break;
+                        case IonTypes.TIMESTAMP :
+                            break;
+                        case IonTypes.CLOB :
+                            tempBuf = [];
+                            for(let i = 0; i < tempEvent.ionValue.length; i++){
+                                tempBuf.push(tempEvent.ionValue.charCodeAt(i))
+                            }
+                            writer.writeClob(tempBuf, tempEvent.annotations);
+                            break;
+                        case IonTypes.BLOB :
+                            tempBuf = [];
+                            for(let i = 0; i < tempEvent.ionValue.length; i++){
+                                tempBuf.push(tempEvent.ionValue.charCodeAt(i))
+                            }
+                            writer.writeBlob(tempBuf, tempEvent.annotations);
+                            break;
+                        default :
+                            throw new Error("unexpected type: " + tempEvent.ionType.name);
+                    }
+                    break;
+                case IonEventType.CONTAINER_START:
+                    switch(tempEvent.ionType) {
+                        case IonTypes.STRUCT :
+                            writer.writeStruct(tempEvent.annotations, false);//need to change.
+                            break;
+                        case IonTypes.LIST :
+                            writer.writeList(tempEvent.annotations, false);
+                            break;
+                        case IonTypes.SEXP :
+                            writer.writeSexp(tempEvent.annotations, false);
+                            break;
+                    }
+
+                case IonEventType.CONTAINER_END:
+                    writer.endContainer();
+                    break;
+                case IonEventType.STREAM_END:
+                    writer.close();
```

Comment:
It seems like close doesn't necessarily do anything IO wise since its just writing to a buffer(no hooks into the OS)
It simply calls close container so its effectively a no op if the event stream input is valid ion.

    close() : void {
        while (!this.isTopLevel) {
            this.endContainer();
        }
    }

Reply:
I made a comment about this `close` implementation in my other review too. I'll also note here that (once you add support for symbol tables and the IVM), it will be expected that close/finish resets the local symbol table context. Something to keep in mind.

Reply:
noted https://github.com/amzn/ion-js/issues/129